### PR TITLE
Add: Intro tooltips for shipping smart defaults

### PR DESCRIPTION
**Pull request by rjchow**
Part 1 of #33365, which introduces the tooltips for each of the designated user flows

Another PR will follow this to update the spotlight to highlight more specific parts of the UI.

### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

A Tour kit flow has been introduced for users who have the shipping smart defaults automatically populated.

### How to test the changes in this Pull Request:

There are 4 separate flows in this PR, same as in #33547 
These instructions extend the ones from that PR

I made a build available here: https://github.com/woocommerce/woocommerce/actions/runs/2613751241

**Case 1: The store sells physical products and is located in the US, but JP and WCS are not installed.**

1. Start with a fresh install and enable the `shipping-smart-defaults` and `shipping-setting-tour` feature flags
2. Start OBW and choose `United States` as store country
3. Choose "Physical products" 
4. Complete the OBW without installing anything from the `Business Details` tab.
5. Navigate to WooCommerce -> Settings -> Shipping
6. "United States (US)" zone should be created with `Free shipping` method
7. **There should be 3 steps in the tour, covering shipping zones, shipping methods and recommended shipping options**
8. The tour should not appear again upon revisit or refresh of the page, if you complete the tour

**Case 2: The store sells physical products, has JP and WCS installed and connected, and is located in the US.**

1. Start with a fresh install and enable the `shipping-smart-defaults` and `shipping-setting-tour` feature flags (alternatively just go and delete the option for `woocommerce_admin_reviewed_default_shipping_zones`)
2. Start OBW and choose `United States` as store country
3. Choose "Physical products" 
4. Install Jetpack and WooCommerce Shipping from the `Business Details` tab.
5. Complete the OBW 
6. Connect and approve Jetpack when prompted.
7. Navigate to WooCommerce -> Settings -> Shipping
9. "United States (US)" zone should be created with `Free shipping` method
8. **There should be 3 steps in the tour, with the third one highlighting the WooCommerce Shipping section**
9. The tour should not appear again upon revisit or refresh of the page, if you complete the tour

**Case 3: The store sells physical products, and is not located in US/Canada/Australia/UK (irrelevant if JP is installed or not).**

1. Start with a fresh install and enable the `shipping-smart-defaults` and `shipping-setting-tour` feature flags
2. Start OBW and choose a country that is not one of the US, Canada, Australia, and UK
3. Choose "Physical products" 
4. Complete the OBW without installing anything from the `Business Details` tab.
5. Navigate to WooCommerce -> Settings -> Shipping
6. "Country-you-chose" zone should be created with `Free shipping` method
7. **There should be 2 steps in the tour, covering shipping zones and shipping methods**
8. The tour should not appear again upon revisit or refresh of the page, if you complete the tour

**Case 4: The store sells digital products**
1. Start with a fresh install and enable the `shipping-smart-defaults` and `shipping-setting-tour` feature flags
2. Start OBW and choose any country
3. Choose "Downloads" 
4. Complete the OBW without installing anything from the `Business Details` tab.
5. Navigate to WooCommerce -> Settings -> Shipping
6. You should not see any automatically created shipping settings and there should not be a tour

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33688/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `add: intro tooltips for shipping smart defaults` 2ae0ac2315327e7fbf602fb401c2fe0d073ef7c8
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: plugins/woocommerce-admin/client/shipping/shipping-tour.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	};
+};
+
+export const ShippingTour = () => {
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+	const { show: showTour } = useShowShippingTour();
+
+	const tourConfig: TourKitTypes.WooConfig = {
+		placement: 'auto',
+		steps: [
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/shipping/shipping-tour.tsx:57]: > There should be 3 steps in the tour, covering shipping zones, shipping methods and recommended shipping options

@rjchow I was not able to see the 3 steps. I think there are only 2 steps in this PR?

Comment (rjchow) [reply plugins/woocommerce-admin/client/shipping/shipping-tour.tsx:57]: Silly me I forgot to push the latest!! Fixed

File: plugins/woocommerce-admin/client/shipping/shipping-tour.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	const WCS_LINK_SELECTOR = 'a[href*="woocommerce-services-settings"]';
+	const isWcsSectionPresent = document.querySelector( WCS_LINK_SELECTOR );
+
+	const SHIPPING_RECOMMENDATIONS_SELECTOR =
+		'div.woocommerce-recommended-shipping-extensions';
+	const isShippingRecommendationsPresent = ! activePlugins.includes(
+		'woocommerce-services'
+	);
+
+	if ( isWcsSectionPresent ) {
```
Comment (adrianduffell) [inline plugins/woocommerce-admin/client/shipping/shipping-tour.tsx:118]: I think in the future, we might want to have this code provided by the WooCommerce Shipping plugin itself. That way the WooCommerce Shipping plugin can iterate on it directly without waiting on the Core release train, for example if the plugin name changes again in the future.

So maybe a future next step is to bring in an integration point for the Shipping plugin (or any other shipping plugin?) to add their banner and tour step in a standardized way.

**Review by adrianduffell — APPROVED** [review]
Nice work on this @rjchow! Things tested well for me overall and LGTM 🚀

File: plugins/woocommerce-admin/client/shipping/shipping-tour.tsx (hunk 2 above)
Comment (rjchow) [reply plugins/woocommerce-admin/client/shipping/shipping-tour.tsx:118]: Good idea yea I think we should make it standard to have a filter for adding tour kit steps

**Comment by chihsuan** [comment]
@rjchow Could you add `woocommerce_admin_created_default_shipping_zones` and `woocommerce_admin_reviewed_default_shipping_zones` to [`$woocommerce_permissions` list](
https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/API/Options.php#L146-L184) in your another PR? 

Otherwise, a user would see a notice `The Automattic\WooCommerce\Admin\API\Options::get_options function is deprecated since version 3.1.` in the debug log. Thanks!

**Comment by rjchow** [comment]
Thanks! I didn't know that!

**Pull request merged**